### PR TITLE
Remove Promise usage from 3P as we removed the polyfill.

### PR DESCRIPTION
**Pull request by cramforce**
Adds a presubmit that checks for the removed polyfills and
also checks the actually build binary in case the transitive
closure gets bigger and we fail to manually include those
files for the 3p test.

Removes some polyfills that are present in our entire browser matrix.

### CODE REVIEW

File: 3p/polyfills.js [hunk 1]
```diff
@@ -21,3 +21,5 @@
 
 // This list should not get longer without a very good reason.
 import 'babel-core/external-helpers';
+
+require('core-js/modules/es6.array.from');
```
Comment (erwinmombay) [inline 3p/polyfills.js]: for knowledge purposes, where is this used?

Comment (cramforce) [inline 3p/polyfills.js]: babel-core/external-helpers

Which we might need to fork :(

It has all kinds of shit for stuff we don't actually use.

Probably better to take it now and switch to closure compiler.

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -64,6 +56,18 @@ var forbiddenTerms = {
...
 
+var ThreePTermsMessage = 'The 3p bootstrap iframe has no polyfills loaded and' +
+    ' can thus not use most modern web APIs.';
+
+var forbidden3pTerms = {
+  // We need to forbid promise usage because we don't have our own polyfill
+  // available. This whitelisting of callNext is a major hack to allow one
+  // usage in babel's external helpers that is in a code path that we do
+  // not use.
+  '\\.then\\((?!callNext)': ThreePTermsMessage,
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: so we can `probably` generate a babelhelpers file that wouldn't include this piece of code. might also trim the babelhelpers file. i tried doing it originally but settled on not doing it as we were still using a lot of stuff heavily. I'll look into it again. (https://babeljs.io/docs/advanced/external-helpers/ , see --whitelist and selective builds)

Comment (cramforce) [inline build-system/tasks/presubmit-checks.js]: Filed #652

Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: thanks.

**Comment by erwinmombay** [comment]
LGTM

**Pull request merged**